packages/mobx/src/core/atom.ts

```diff
@@ -68,21 +68,19 @@ export class Atom implements IAtom {
      * Invoke this method _after_ this method has changed to signal mobx that all its observers should invalidate.
      */
     public reportChanged() {
-        if (globalState.inBatch && this.batchId_ === globalState.batchId) {
-            // Called from the same batch this atom was created in.
-            return
+        if (globalState.inBatch && this.batchId_ !== globalState.batchId) {
```

Comment:
`!globalState.inBatch || this.batchId_ !== globalState.batchId`
If not in batch mutation always updates version immediately. 
If in batch only if it's different.